opensearch-aws-sigv4/lib/opensearch-aws-sigv4.rb

```diff
@@ -50,7 +50,12 @@ def initialize(transport_args = {}, sigv4_signer, &block)
 
       # @see OpenSearch::Transport::Transport::Base::perform_request
       def perform_request(method, path, params = {}, body = nil, headers = nil)
-        verify_open_search unless @verified
+        unless @verified
+          verify_signature = sigv4_signer.sign_request(
+            http_method: 'GET',
+            url: signature_url('/', {}))
+          verify_open_search(verify_signature.headers)
+        end
```

Comment:
Refactor this. 

1. Implement `verify_open_search` as a private method in this class in which you do the `verify_signature = `..., then call `super`. 
1. Same for below, instead of calling `@transport.perform_request`, call `super` after generating the signature.

Reply:
@dblock thanks for your feedback, I've updated the PR with your suggested refactoring.